src/codegen/codegen_c_visitor.cpp

```diff
@@ -3736,6 +3746,49 @@ void CodegenCVisitor::print_net_send_buffering() {
 }
 
 
+void CodegenCVisitor::visit_for_netcon(ast::ForNetcon& node) {
+    // For_netcon should take the same arguments as net_receive and apply the operations
+    // in the block to the weights of the netcons. Since all the weights are on the same vector,
+    // weights, we have a mask of operations that we apply iteratively, advancing the offset
+    // to the next netcon.
+    const auto& args = node.get_parameters();
+    RenameVisitor v;
+    auto& statement_block = node.get_statement_block();
+    for (size_t i_arg = 0; i_arg < args.size(); ++i_arg) {
+        // sanitize node_name since we want to substitute names like (*w) as they are
+        auto old_name =
+            std::regex_replace(args[i_arg]->get_node_name(), regex_special_chars, R"(\$&)");
+        auto new_name = "weights[{} + nt->_fornetcon_weight_perm[i]]"_format(i_arg);
+        v.set(old_name, new_name);
+        statement_block->accept(v);
+    }
```

Comment:
Just curious -> if we transform this FOR_NETCON block before visiting NET_RECEIVE, would it avoid this regex search/replace stuff?

what I meant is : in the visit_net_receieve, before going to children's, fix FOR_NETCON first if exist.

Reply:
mmm I am not sure how would you do this. In print_net_init we rename the variables. Then there are other functions that print out changed variables like print_watch_activate(). Then we visit for_netcon and print it. If we print for_netcon first we did not call print_watch_activate or we did print with the wrong names (if we replace after fornetcon)